build/jobs/analyze.yml

```diff
@@ -0,0 +1,107 @@
+parameters:
+  analyzeARMTemplates: true
+  analyzeBinaries: true
+  analyzePackages: true
+  runAntiMalware: true
+  credScanDirectory: '$(Build.SourcesDirectory)'
+
+steps:
+- ${{ if eq(parameters.analyzeBinaries, 'true') }}:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Binaries'
+    inputs:
+      buildType: 'current'
+      downloadType: 'single'
+      downloadPath: '$(Agent.TempDirectory)/artifacts'
+      artifactName: 'deploy'
+
+- ${{ if eq(parameters.analyzePackages, 'true') }}:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download NuGet Packages'
+    inputs:
+      buildType: 'current'
+      downloadType: 'single'
+      downloadPath: '$(Build.SourcesDirectory)/artifacts'
+      artifactName: 'nuget'
+
+- ${{ if eq(parameters.analyzeBinaries, 'true') }}:
+  - task: ExtractFiles@1
+    displayName: 'Extract Stu3 Web Server Binaries'
+    inputs:
+      archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.Stu3.Web.zip'
+      destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/Stu3'
+  - task: ExtractFiles@1
+    displayName: 'Extract R4 Web Server Binaries'
+    inputs:
+      archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.R4.Web.zip'
+      destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/r4'
+  - task: ExtractFiles@1
+    displayName: 'Extract R5 Web Server Binaries'
+    inputs:
+      archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.R5.Web.zip'
+      destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/r5'
+
+- ${{ if eq(parameters.runAntiMalware, 'true') }}:
+  - task: AntiMalware@4
+    inputs:
+      InputType: 'Basic'
+      ScanType: 'CustomScan'
+      FileDirPath: '$(Build.SourcesDirectory)'
+      EnableServices: true
+      TreatSignatureUpdateFailureAs: 'Standard'
+      SignatureFreshness: 'OneDay'
+      TreatStaleSignatureAs: 'Error'
+
+- ${{ if eq(parameters.analyzeARMTemplates, 'true') }}:
+  - task: Armory@2
+    inputs:
+      targetDirectory: '$(Build.SourcesDirectory)\samples\templates'
+      targetFiles: 'f|*.json'
+      excludePassesFromLog: false
+
+- task: CredScan@3
+  inputs:
+    scanFolder: ${{ parameters.credScanDirectory }}
```

Comment:
Why does CredScan need a parameter for its scan folder? All the other analyzers in the files just assume Build.SourcesDirectory.

Reply:
Looks like it could be used if we split out some new pipelines for sub-areas:
![image](https://user-images.githubusercontent.com/197221/189711755-d7b48ab6-27dd-460c-9a68-c62fb75bea7d.png)